tools/mo/openvino/tools/mo/utils/cli_parser.py

```diff
@@ -210,38 +159,109 @@ def single_input_to_str(input):
                 if name is not None:
                     raise Exception("More than one input name provided: {}".format(input))
                 name = val
-            elif isinstance(val, type) or isinstance(val, Type):
+            elif isinstance(val, (type, Type)):
                 if inp_type is not None:
                     raise Exception("More than one input type provided: {}".format(input))
-                inp_type = type_to_str(val)
+                inp_type = val
             elif is_shape_type(val):
                 if shape is not None:
                     raise Exception("More than one input shape provided: {}".format(input))
-                shape = shape_to_str(val, " ")
+                shape = PartialShape(val)
             else:
-                raise Exception("Incorrect input parameters provided. Expected input name and "
-                                "optionally input type or input shape. Got unknown object: {}".format(val))
-        if name is None:
-            raise Exception("Input name was not provided for following input {}.".format(input))
-        if shape is not None:
-            name += shape
-        if inp_type is not None:
-            name += "{" + inp_type + "}"
-        return name
+                raise Exception("Incorrect input parameters provided. Expected tuple with input name, "
+                                "input type or input shape. Got unknown object: {}".format(val))
+        return openvino.tools.mo.InputCutInfo(name,
+                                              PartialShape(shape) if shape is not None else None,
+                                              inp_type,
+                                              None)
+    if isinstance(input, (type, Type)):
+        return openvino.tools.mo.InputCutInfo(None, None, input, None)
 
     raise Exception("Unexpected object provided for input. Expected openvino.tools.mo.InputCutInfo "
                     "or tuple or str. Got {}".format(type(input)))
 
 
-def input_to_str(input):
-    if input is None or isinstance(input, str):
-        return input
+def input_to_input_cut_info(input):
```

Comment:
can you please provide description for this function?

Reply:
and for other functions as well

Reply:
Done.